MC/config/PWGHF/external/generator/generator_pythia8_gaptriggered_hf.C

```diff
@@ -161,19 +216,23 @@ private:
   int mInverseTriggerRatio;
 
   // Control alternate injection of charm and beauty
-  bool mDoAltInjection;
-  std::vector<int> mQuarkPdgList;
+  std::vector<int> mQuarkPdgList = {};
+
+  // Control alternate injection of Omega and Xi
+  std::vector<int> mHadronPdgList = {};
 };
 
 // Predefined generators:
 // Charm-enriched
-FairGenerator *GeneratorPythia8GapTriggeredCharm(int inputTriggerRatio, float yQuarkMin=-1.5, float yQuarkMax=1.5, float yHadronMin=-1.5, float yHadronMax=1.5, int pdgCodeCharmHadron=0, bool doNoQuarkTrigger=false) {
-  auto myGen = new GeneratorPythia8GapTriggeredHF(inputTriggerRatio, std::vector<int>{4}, false);
+FairGenerator *GeneratorPythia8GapTriggeredCharm(int inputTriggerRatio, float yQuarkMin = -1.5, float yQuarkMax = 1.5, float yHadronMin = -1.5, float yHadronMax = 1.5, int pdgCodeCharmHadron = 0, bool doNoQuarkTrigger = false)
```

Comment:
Since you added a dedicated function that allows not to have a trigger at the parton level, I would remove the option from here
```suggestion
FairGenerator *GeneratorPythia8GapTriggeredCharm(int inputTriggerRatio, float yQuarkMin = -1.5, float yQuarkMax = 1.5, float yHadronMin = -1.5, float yHadronMax = 1.5, std::vector<int> pdgCodeCharmHadron = {})
```